worker/methods.go

```diff
@@ -238,6 +238,52 @@ func (w *Worker) TaskList() (tasks []TaskInfo, err error) {
 	return out["tasks"], nil
 }
 
+type TaskListParams struct {
+	CodeName string
+	Page     int
+	PerPage  int
+	FromTime time.Time
+	ToTime   time.Time
+	Statuses []string
+}
+
+func (w *Worker) FilteredTaskList(params TaskListParams) (tasks []TaskInfo, err error) {
+	out := map[string][]TaskInfo{}
+	url := w.tasks()
+
+	if params.CodeName != "" {
```

Comment:
This looks like `code_name` parameter, represented by `TaskListParams.CodeName`, is optional API parameter. Could you please remove the check, that `CodeName` is empty, and keep just `url.QueryAdd(...)`.